lib/omniauth/strategies/saml.rb

```diff
@@ -10,7 +10,7 @@ def self.inherited(subclass)
         OmniAuth::Strategy.included(subclass)
       end
 
-      OTHER_REQUEST_OPTIONS = [:skip_conditions, :allowed_clock_drift, :matches_request_id, :skip_subject_confirmation].freeze
+      OTHER_REQUEST_OPTIONS = [:skip_conditions, :allowed_clock_drift, :matches_request_id, :skip_subject_confirmation, :soft].freeze
```

Comment:
Agreed, this should not be here. It's not necessary, since the `soft` option is already passed to `ruby-saml` in the `settings` hash.